src/frequenz/sdk/microgrid/_power_managing/_matryoshka.py

```diff
@@ -199,7 +199,15 @@ def calculate_target_power(
             bucket = self._component_buckets.setdefault(component_ids, set())
             if proposal in bucket:
                 bucket.remove(proposal)
-            bucket.add(proposal)
+            if (
+                proposal.preferred_power is not None
+                or proposal.bounds.lower is not None
+                or proposal.bounds.upper is not None
+            ):
+                bucket.add(proposal)
+            elif not bucket:
+                del self._component_buckets[component_ids]
+                _ = self._target_power.pop(component_ids, None)
```

Comment:
I don't think you need the `_ = `. Also is it a `pop` instead of a `del` because `component_id` might not be in `self._target_power`?

Reply:
Which linter is that? I did a quick search and didn't found any linter that has that feature. I agree is a good lint to have, otherwise it looks very arbitrary, as we have many other places where values are silently discarded. It would be nice to enabled it for everyone so we don't end up with inconsistent code (that it might also be very noisy for the people using the pendantic tool).

Reply:
I've been using https://docs.basedpyright.com/latest/ and enjoying it a lot.  I've been using it as an LSP server, but it is also available as a CLI tool.